mypy/meet.py

```diff
@@ -255,10 +255,11 @@ def visit_typeddict_type(self, t: TypedDictType) -> Type:
             for (_, l, r) in self.s.zip(t):
                 if not is_equivalent(l, r):
                     return self.default(self.s)
-            items = OrderedDict([
-                (item_name, s_item_type or t_item_type)
-                for (item_name, s_item_type, t_item_type) in self.s.zipall(t)
-            ])
+            item_list = cast(List[Tuple[str, Type]],
```

Comment:
I would choose the second option (for-loop), the comprehension is a bit hard to read here.